### PR TITLE
feat: add config for mode on file creation

### LINKED ISSUE #129: Allow users to specify the creating file mode
Files created by sonic-boom and pino have the default of `0o666` as every other file created by Node.js: https://nodejs.org/api/fs.html#fsopenpath-flags-mode-callback.

We should support making this configurable.

### PR DESCRIPTION
**Pull request by estherixz**
Fixes #129

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -670,6 +670,108 @@ function buildTests (test, sync) {
       }, 0)
     })
   })
+
+  test('mode', (t) => {
```
Comment (estherixz) [inline test.js:674]: To test I looked at the current [nodejs tests for fs.open](https://github.com/nodejs/node/blob/master/test/parallel/test-fs-open-mode-mask.js).

Unfortunately those tests do not test extensively passing different modes.

File: test.js [hunk 2]
```diff
@@ -670,6 +670,108 @@ function buildTests (test, sync) {
...
+    t.ok(stream.write('something else\n'))
+
+    stream.end()
+
+    stream.on('finish', () => {
+      fs.readFile(dest, 'utf8', (err, data) => {
+        t.error(err)
+        t.equal(data, 'hello world\nsomething else\n')
+        t.not(fs.statSync(dest).mode & 0o777, stream.mode)
+        t.equal(fs.statSync(dest).mode & 0o777, defaultMode)
```
Comment (estherixz) [inline test.js:746]: This test demonstrates how passing a mode doesn't always change to the expected permissions.

Testing `fs.open` in my local machine I was getting identical results, so this seems to be the way `fs.open` works at the moment.

File: README.md [hunk 3]
```diff
@@ -64,6 +64,7 @@ The options are:
   to exceed `maxLength`, the data written is dropped and a `drop` event is emitted with the dropped data
 * `sync`: perform writes synchronously (similar to `console.log`).
 * `append`: appends writes to dest file instead of truncating it (default `true`).
+* `mode`: specify the creating file `mode`.
```
Comment (mcollina) [inline README.md:67]: Please link `fs.open()` from Node.js core: https://nodejs.org/api/fs.html#fsopenpath-flags-mode-callback

File: test.js [hunk 4]
```diff
@@ -670,6 +675,108 @@ function buildTests (test, sync) {
       }, 0)
     })
   })
+
+  test('mode', (t) => {
```
Comment (mcollina) [inline test.js:679]: Please skip when using windows.

File: test.js [hunk 5]
```diff
@@ -670,6 +675,108 @@ function buildTests (test, sync) {
...
+    stream.on('finish', () => {
+      fs.readFile(dest, 'utf8', (err, data) => {
+        t.error(err)
+        t.equal(data, 'hello world\nsomething else\n')
+        t.equal(fs.statSync(dest).mode & 0o777, stream.mode)
+      })
+    })
+  })
+
+  test('mode default', (t) => {
```
Comment (mcollina) [inline test.js:705]: Please skip when using windows.

File: test.js [hunk 6]
```diff
@@ -670,6 +675,108 @@ function buildTests (test, sync) {
...
+    stream.on('finish', () => {
+      fs.readFile(dest, 'utf8', (err, data) => {
+        t.error(err)
+        t.equal(data, 'hello world\nsomething else\n')
+        t.equal(fs.statSync(dest).mode & 0o777, defaultMode)
+      })
+    })
+  })
+
+  test('mode on mkdir', (t) => {
```
Comment (mcollina) [inline test.js:730]: Please skip when using windows.

File: test.js [hunk 7]
```diff
@@ -670,6 +675,108 @@ function buildTests (test, sync) {
...
+      fs.readFile(dest, 'utf8', (err, data) => {
+        t.error(err)
+        t.equal(data, 'hello world\n')
+        t.equal(fs.statSync(dest).mode & 0o777, stream.mode)
+        stream.end()
+      })
+    })
+  })
+
+  test('mode on append', (t) => {
```
Comment (mcollina) [inline test.js:755]: Please skip when using windows.

File: test.js (hunk 4 above)
Comment (estherixz) [reply test.js:679]: Oh... my "fix" for windows didn't work 😞 
Yes I'll skip!

File: test.js (hunk 2 above)
Comment (estherixz) [reply test.js]: Thank you this helped a lot, I was quite confused! 😅

File: test.js (hunk 4 above)
Comment (mcollina) [reply test.js:679]: None of this is really supported on Windows. It's the unix permission system. Use the `skip` option of tap's `test` function.

Comment (estherixz) [reply test.js]: yup makes sense! :)

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
@jsumners this is ready to go. I think we can land this safely, would you like to take a quick look?

**Review by jsumners — APPROVED** [review]

**Pull request merged**